src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -106,31 +111,53 @@ export class BrowsingContextProcessor {
           `referenceContext should be a top-level context`
         );
       }
+      userContext = referenceContext.userContext;
     }
 
-    let result: Protocol.Target.CreateTargetResponse;
-
+    let newWindow = false;
     switch (params.type) {
       case BrowsingContext.CreateType.Tab:
-        result = await this.#browserCdpClient.sendCommand(
-          'Target.createTarget',
-          {
-            url: 'about:blank',
-            newWindow: false,
-          }
-        );
+        newWindow = false;
         break;
       case BrowsingContext.CreateType.Window:
-        result = await this.#browserCdpClient.sendCommand(
-          'Target.createTarget',
-          {
-            url: 'about:blank',
-            newWindow: true,
-          }
-        );
+        newWindow = true;
         break;
     }
 
+    if (userContext !== 'default') {
+      const existingContexts = this.#browsingContextStorage
+        .getAllContexts()
+        .filter((context) => context.userContext === userContext);
+
+      if (!existingContexts.length) {
+        // If there are no contexts in the given user context, we need to set
+        // newWindow to true as newWindow=false will be rejected.
+        newWindow = true;
+      }
+    }
+
+    let result: Protocol.Target.CreateTargetResponse;
+
+    try {
+      result = await this.#browserCdpClient.sendCommand('Target.createTarget', {
+        url: 'about:blank',
+        newWindow,
+        browserContextId: userContext === 'default' ? undefined : userContext,
+      });
+    } catch (err) {
+      // https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/devtools/protocol/target_handler.cc;l=1;drc=e80392ac11e48a691f4309964cab83a3a59e01c8
+      if (
+        (err as Error).message.startsWith(
+          'Failed to find browser context with id'
+        )
+      ) {
+        throw new NoSuchUserContextException(
+          `The context ${userContext} was not found`
+        );
+      }
+      throw err;
```

Comment:
consider extracting into a method

Reply:
only the error handling part of the entire create target part?